src/cmd/build.js

```diff
@@ -124,11 +130,42 @@ async function defaultPackageCreator({
   const packageName = safeFileName(
     `${extensionName}-${manifestData.version}.zip`);
   const extensionPath = path.join(artifactsDir, packageName);
-  const stream = createWriteStream(extensionPath);
+  let destStat;
+  try {
+    destStat = await fs.stat(extensionPath);
+  } catch (error) {
+    destStat = false;
+    throw error;
+  }
+  const destinationFileExists = destStat && destStat.isFile();
+
+  function throwExtensionExists() {
+    throw new UsageError(
+      `Extension exists at the destination path: ${extensionPath}\n` +
+      'Use --overwrite-dest to enable overwriting.');
+  }
+
+  if (destinationFileExists) {
+    if (overwriteDest) {
```

Comment:
Instead of nesting two `if` statement into each other we can probably do something like:

```js
if (!overwriteDest && destinationFileExists) {
  throw new UsageError(...);
}

if (destinationFileExists) {
  log.info(`Destination exists, ...`);
}

...
```